torchgeo/datasets/inria.py

```diff
@@ -0,0 +1,243 @@
+    * Labels - RGB aerial GeoTIFFs of shape 5000 x 5000
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://doi.org/10.1109/IGARSS.2017.8127684
+
+    .. versionadded:: 0.3
+    """
+
+    directory = "AerialImageDataset"
+    filename = "NEW2-AerialImageDataset.zip"
+    md5 = "4b1acfe84ae9961edc1a6049f940380f"
+
+    def __init__(
+        self,
+        root: str,
+        split: str = "train",
+        transforms: Optional[Callable[[Dict[str, Any]], Dict[str, Any]]] = None,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new InriaAerialImageLabeling Dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            split: train/test split
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version.
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            RuntimeError: if dataset is missing
+        """
+        self.root = root
+        assert split in {"train", "test"}
+        self.split = split
+        self.transforms = transforms
+        self.checksum = checksum
+
+        self._check_integrity()
+        self.files = self._load_files(root)
+
+    def _load_files(self, root: str) -> List[Dict[str, str]]:
+        """Return the paths of the files in the dataset.
+
+        Args:
+            root: root dir of dataset
+
+        Returns:
+            list of dicts containing paths for each pair of image and label
+        """
+        files = []
+        root_dir = os.path.join(root, self.directory, self.split)
+        images = glob.glob(os.path.join(root_dir, "images", "*.tif"))
+        images = sorted(images)
+        if self.split == "train":
+            labels = glob.glob(os.path.join(root_dir, "gt", "*.tif"))
+            labels = sorted(labels)
+
+            for img, lbl in zip(images, labels):
+                files.append({"image_path": img, "label_path": lbl})
+        else:
+            for img in images:
+                files.append({"image_path": img})
+
+        return files
+
+    def _load_image(self, path: str) -> Tensor:
+        """Load a single image.
+
+        Args:
+            path: path to the image
+
+        Returns:
+            the image
+        """
+        filename = os.path.join(path)
+        with rio.open(filename) as img:
+            array = img.read().astype(np.int32)
+            tensor: Tensor = torch.from_numpy(array)  # type: ignore[attr-defined]
+            return tensor
+
+    def _load_target(self, path: str) -> Tensor:
+        """Loads the target mask.
+
+        Args:
+            path: path to the mask
+
+        Returns:
+            the target mask
+        """
+        filename = os.path.join(path)
+        with rio.open(filename) as img:
+            array = img.read().astype(np.int32)
+            array = np.clip(array, 0, 1)
```

Comment:
This is quite different than dividing by 255. This means that all values greater than 1 will be changed to 1. Can you try plotting again with the latest code and make sure this is what you want?

Reply:
The mask is binary but the value for building is set to 255 instead of 1. Any value greater than 0 should be set to 1.

Reply:
Ah gotcha, thanks for clarifying!